### ISSUE TITLE
Extract registration name from function and class

### ISSUE THREAD

**Issue by mikhas-rock** [jeffijoe/awilix#26]
It would prevent typing the names twice in several occasions:

```javascript
class MyClass { /* ... */}
function myFunction(MyClass){ /* ... */ }

// AS IS
container.registerClass('MyClass', MyClass);
container.registerFunction('myFunction', myFunction);

// SUGGESTION
container.registerClass(MyClass);
container.registerFunction(myFunction);

container.register(MyClass);
container.register(myFunction);

**Comment by jeffijoe** [comment]
I like the suggestion, but in case `.name` does not _actually_ return what you expect, you wouldn't know what it was registered as. For example:

```js
const myFunction = function myAwesomeFunction () {}

container.registerFunction(myFunction)

container.resolve('myFunction') // boom
```

Additionally, it can only work for `registerClass` and `registerFunction`, not `register` (because you don't register actual values with `register`, you use the `asClass`/`asFunction` chains)

Thoughts?

**Comment by mikhas-rock** [comment]
Agreed: this behavior could be applied only on registerClass and registerFunction.

**Comment by jeffijoe** [comment]
Implemented and released as 2.6.0. Thanks!
